### PR TITLE
Add public getters for CircularBuffer sizes on BufferedSerial

### PR DESCRIPTION
**Pull request by qwertyquerty**
### Summary of changes 

Adds `size_t txbuf_size() const;` and `size_t rxbuf_size() const;` to expose the current usage of each buffer

Changed counter type to 16 bits to allow larger buffer sizes

### Documentation 

None

----------------------------------------------------------------------------------------------------------------
### Pull request type
    [] Patch update (Bug fix / Target update / Docs update / Test update / Refactor)
    [x] Feature update (New feature / Functionality change / New API)
    [] Major update (Breaking change E.g. Return code change / API behaviour change)

----------------------------------------------------------------------------------------------------------------
### Test results

    [] No Tests required for this change (E.g docs only update)
    [] Covered by existing mbed-os tests (Greentea or Unittest)
    [] Tests / results supplied as part of this PR
    
    
----------------------------------------------------------------------------------------------------------------
### Reviewers 

@multiplemonomials 
----------------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by qwertyquerty** [comment]
Squash on merge

**Comment by JohnK1987** [comment]
You wrote

> Adds size_t txbuf_size() const; and size_t rxbuf_size() const; to expose the current usage of each buffer

> Changed counter type to 16 bits to allow larger buffer sizes

I understand to first point but you changed data type from size_t to uint16_t, why?

The second point is not clear to me. The constructor of CircularBuffer has 3 parameters.
1. Data Type 
2. Size of buffer
3. Counter Type = uint32_t (default value)

You propose to change first parameter but how you can see, it is Data type of the buffer and not type of the Counter which is already set to uint32_t.

**Comment by JohnK1987** [comment]
@qwertyquerty any comments?

File: drivers/include/drivers/BufferedSerial.h [hunk 1]
```diff
@@ -327,8 +342,8 @@ class BufferedSerial:
      *  By default buffer size is 256 for TX and 256 for RX. Configurable
      *  through mbed_app.json
      */
-    CircularBuffer<char, MBED_CONF_DRIVERS_UART_SERIAL_RXBUF_SIZE> _rxbuf;
-    CircularBuffer<char, MBED_CONF_DRIVERS_UART_SERIAL_TXBUF_SIZE> _txbuf;
+    CircularBuffer<uint16_t, MBED_CONF_DRIVERS_UART_SERIAL_RXBUF_SIZE> _rxbuf;
+    CircularBuffer<uint16_t, MBED_CONF_DRIVERS_UART_SERIAL_TXBUF_SIZE> _txbuf;
```
Comment (JohnK1987) [inline drivers/include/drivers/BufferedSerial.h:346]: This parametr for is for data type of buffer and not for Counter type. The counter typer is third parametr and what is not used in this case and that apply its default value uint32_t.
 so kind and revert this change back.

**Comment by JohnK1987** [comment]
I close this because there is no more activity from autor for for more than week.
